### PR TITLE
refactor(*): first pass at turning class components into functional components

### PR DESCRIPTION
**Pull request by ramfox**
- bumps version to 0.7.2-dev
- changes class components that do not use lifecycle components into functional components
  - item/CitationItem
  - item/ContributorItem
  - item/DatasetItem
  - item/FieldItem
  - item/KeyValueItem
  - item/ProfileItem
  - item/SearchItem
  - item/StatItem
  - item/TagItem
  - chrome/Button
  - chrome/Dropdown
  - chrome/DropdownMenu
  - chrome/NavLinks
  - chrome/Spinner
  - chrome/TopBar
- removes the app/dist contents, which should have been being ignored based on the .gitignore file

### CODE REVIEW

File: lib/components/chrome/Button.js [hunk 1]
```diff
@@ -3,58 +3,71 @@ import { Link } from 'react-router-dom'
...
-  downloadName: PropTypes.string,
-  color: PropTypes.string,
-  onClick: PropTypes.func,
-  full: PropTypes.bool,
-  attached: PropTypes.bool,
-  loading: PropTypes.bool,
-  download: PropTypes.string,
-  link: PropTypes.string,
-  disabled: PropTypes.bool
+  text: PropTypes.string.isRequired, // the text written on the button
```
Comment (b5) [inline lib/components/chrome/Button.js:57]: In some of these prop comments we have comment inline, others comment above-line. What say you to always writing comment above-line? It’ll keep consistent with our go code

File: lib/components/chrome/Button.js [hunk 2]
```diff
@@ -3,58 +3,71 @@ import { Link } from 'react-router-dom'
...
-    }
-
-    if (download && downloadName) {
-      options['href'] = download
-      options['download'] = download ? `${downloadName}.zip` : undefined
-      return (
-        <a {...options}>
+
+// Button is a the basic button used throughout the app
+const Button = ({ color,
```
Comment (b5) [inline lib/components/chrome/Button.js:8]: I prefer a variable `props` argument that is then destructured on the next line:

```js
const Button = (props) => {
  const { color, onClick /* ... */ } = props
  // ...
}
```

To me it’s clearer what’s going on when switching between functional and Class components. Thoughts?

**Review by b5 — APPROVED** [review]
Looking great. This is a huge improvement. A few bits then LGTM

File: lib/components/chrome/Button.js (hunk 2 above)
Comment (ramfox) [reply lib/components/chrome/Button.js:8]: sounds good! makes sense! changing it now :)

File: lib/components/chrome/Button.js (hunk 1 above)
Comment (ramfox) [reply lib/components/chrome/Button.js:57]: Yes, thank you for catching this

**Pull request merged**